lib/spack/spack/buildcache_prune.py

```diff
@@ -106,22 +113,29 @@ def _delete_manifests_from_cache_aws(
     return None
 
 
-def _delete_entries_from_cache_manual(url: str, urls_to_delete: Set[str]) -> int:
+def _delete_entries_from_cache_manual(
+    url: str, urls_to_delete: Set[str], pruning_started_at: float
+) -> int:
     pruned_objects = 0
     futures: List[Future] = []
 
     with spack.util.parallel.make_concurrent_executor() as executor:
         for url in urls_to_delete:
-            futures.append(executor.submit(_delete_object, url))
+            futures.append(executor.submit(_delete_object, url, pruning_started_at))
 
         for manifest_or_blob_future in as_completed(futures):
             pruned_objects += manifest_or_blob_future.result()
 
     return pruned_objects
 
 
-def _delete_object(url: str) -> int:
+def _delete_object(url: str, pruning_started_at: float) -> int:
     try:
+        stat_result = web_util.stat_url(url)
+        assert stat_result is not None
+        if stat_result[1] > pruning_started_at:
+            tty.info(f"Skipping deletion of {url} because it was modified after pruning started")
```

Comment:
This can potentially output a lot, and this case is non-destructive so I don't think reporting it by default is necessarily required.

```suggestion
            tty.verbose(f"Skipping deletion of {url} because it was modified after pruning started")
```